.github/workflows/publish.yml

```
@@ -0,0 +1,17 @@
+name: Publish
+
+on:
+    push:
+        branches:
+            - main
+
+jobs:
+    publish:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            id-token: write
+        steps:
+            - uses: actions/checkout@v4
+            - uses: denoland/setup-deno@v1
+            - run: deno publish
```

We should publish deno if we create a github release. We can run `deno publish --dry-run` when pushing to main branch.
And for the workspaces feature we would need to enable `DENO_UNSTABLE_WORKSPACES`.

```suggestion
name: Publish

on:
  push:
    branches: [main]
  release:
    types: [published]

env:
  DENO_UNSTABLE_WORKSPACES: true

jobs:
  publish:
    runs-on: ubuntu-latest

    permissions:
      contents: read
      id-token: write

    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Set up Deno
        uses: denoland/setup-deno@v1

      - name: Publish (dry run)
        if: github.event_name == 'push'
        run: deno publish --dry-run

      - name: Publish
        if: github.event_name == 'release'
        run: deno publish
```